brvs-core/src/test/java/iroha/validation/behavior/ChainAdapterTest.java

```diff
@@ -0,0 +1,167 @@
+package iroha.validation.behavior;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import iroha.protocol.BlockOuterClass;
+import iroha.protocol.Primitive.RolePermission;
+import iroha.validation.adapter.ChainAdapter;
+import iroha.validation.transactions.provider.impl.IrohaHelper;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+import jp.co.soramitsu.crypto.ed25519.Ed25519Sha3;
+import jp.co.soramitsu.iroha.java.IrohaAPI;
+import jp.co.soramitsu.iroha.java.QueryBuilder;
+import jp.co.soramitsu.iroha.java.Transaction;
+import jp.co.soramitsu.iroha.testcontainers.IrohaContainer;
+import jp.co.soramitsu.iroha.testcontainers.PeerConfig;
+import jp.co.soramitsu.iroha.testcontainers.detail.GenesisBlockBuilder;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testcontainers.containers.GenericContainer;
+
+public class ChainAdapterTest {
```

Comment:
Why do we need separate class for that? Just place the test to `IrohaIntegrationTest`